docs/sql-ref-functions-builtin.md

```diff
@@ -0,0 +1,77 @@
+---
+layout: global
+title: Built-in Functions
+displayTitle: Built-in Functions
+license: |
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+  http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+{% for static_file in site.static_files %}
+    {% if static_file.name == 'generated-agg-funcs-table.html' %}
+### Aggregate Functions
+{% include_relative generated-agg-funcs-table.html %}
+#### Examples
+{% include_relative generated-agg-funcs-examples.html %}
```

Comment:
So we put all the examples in one section?

Reply:
Yea, we currently put examples per group in one section  like this;
```
### Aggrgate Functions
<A list of aggregate functions>
#### Examples
<All examples of the aggregate functions listed above>

### Window Functions
<A list of aggregate functions>
#### Examples
<All examples of the window functions listed above>
...
```

Reply:
I think it's better to put everything about one function (signature, desc, examples, etc.) together. Did we decide to do this due to technical difficulties?

Reply:
We can do it. I just a bit rushed to merge this to keep this PR narrow-scoped as a base work. 
Nothing is completely pinned at this moment.

As an easy way, we can refer other DBMSs documentations and mimic I guess.

Reply:
Yea, I think so and we can improve the structure by follow-up activities. But, I'm not sure the improvement should be applied at  the 3.0 release.

Reply:
yea we can improve it in 3.1

Reply:
ok, thanks, I'll file jira later.

Reply:
filed: https://issues.apache.org/jira/browse/SPARK-31513